### PR TITLE
Feature/parallel builds

### PR DESCRIPTION
**Pull request by kwin**
An example build with this library can be seen at https://ci-builds.apache.org/blue/organizations/jenkins/Sling%2Fmodules%2Fsling-maven-enforcer-rules/detail/feature%2Ftest-parallel-builds/29/pipeline

### CODE REVIEW

File: vars/slingOsgiBundleBuild.groovy [hunk 1]
```diff
@@ -122,21 +161,117 @@ def defineStage(def globalConfig, def jobConfig, def jdkVersion, def isReference
...
+    def mavenVersion = mavenPom.version ?: mavenPom.parent.version
+    def isSnapshot = mavenVersion.endsWith('-SNAPSHOT')
+    if ( !isSnapshot ) {
+        return false
+    }
+    return true
+}
+
+boolean isOnMainBranch() {
+    return env.BRANCH_NAME == 'feature/test-parallel-builds'
```
Comment (kwin) [inline vars/slingOsgiBundleBuild.groovy]: Before being merged this needs to be reverted back to `master`

File: vars/slingOsgiBundleBuild.groovy [hunk 2]
```diff
@@ -122,21 +161,117 @@ def defineStage(def globalConfig, def jobConfig, def jdkVersion, def isReference
...
-            invocation.call()
+        wrapInNode(branchConfig.nodeLabel ?: globalConfig.mainNodeLabel, {
+            echo "Running on node ${env.NODE_NAME}"
+            timeout(time: 30, unit: 'MINUTES') {
+                stage("Maven Build (Java ${jdkVersion}, ${goal})") {
+                    invocation.call()
+                }
+            }
+            if ( isReferenceStage ) {
+                // SonarQube (must be wrapped in same node)
```
Comment (rombert) [inline vars/slingOsgiBundleBuild.groovy]: Is this done to ensure that that the artifacts built earlier are used for the SonarQube analysis?

**Review by rombert — APPROVED** [review]
LGTM, thanks!

File: vars/slingOsgiBundleBuild.groovy (hunk 2 above)
Comment (kwin) [reply vars/slingOsgiBundleBuild.groovy]: Yes, exactly. Stashing is too much effort (and it is not clear which files are actually used by the sonar goal)

Comment (rombert) [reply vars/slingOsgiBundleBuild.groovy]: Can then please add this as a comment so it's clear to everyone reading the code?

Comment (kwin) [reply vars/slingOsgiBundleBuild.groovy]: Done in https://github.com/apache/sling-tooling-jenkins/pull/8/commits/54bd0574d8412bd621f4e4372665da5973d30a6a

**Pull request merged**